### DISCUSSION TITLE
How to install this? Can you provide a tutorial?

### DISCUSSION THREAD

**Discussion by xyh2132** [beyondcode/writeout.ai#8] [Q&A]
How to install this? Can you provide a tutorial?

**Comment by laowantong077** [comment]
I have already cloned the git, but I don't know how to open it next. Please help.

**Comment by vmartuniyk** [comment]
@xyh2132  @laowantong077 Guys you must clone  this repository ```git clone https://github.com/beyondcode/writeout.ai``` 
After that go to new directory ``` cd writeout.ai ```
Run few command
```
1.Run composer install
2.Run cp .env.example .env
3.Run php artisan key:generate
4.Run php artisan migrate
5.Run php artisan serve
```
After that added key to env file(read readme file)

**Comment by nxtreaming** [comment]
I got the following errors when I run "php artisan migrate"

root@ecc01:/usr/local/nginx/html/# php artisan migrate

   Illuminate\Database\QueryException

  SQLSTATE[HY000] [2002] No such file or directory (SQL: select * from information_schema.tables where table_schema = transcribe_ai and table_name = migrations and table_type = 'BASE TABLE')

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:760
    756▕         // If an exception occurs when attempting to run a query, we'll format the error
    757▕         // message to include the bindings with SQL, which will make this exception a
    758▕         // lot more helpful to the developer instead of just the database's errors.
    759▕         catch (Exception $e) {
  ➜ 760▕             throw new QueryException(
    761▕                 $query, $this->prepareBindings($bindings), $e
    762▕             );
    763▕         }
    764▕     }

      +39 vendor frames
  40  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()

**Comment by vmartuniyk** [comment]
@nxtreaming in your .env file you have a 3 line
```
DB_DATABASE=transcribe_ai
DB_USERNAME=
DB_PASSWORD=
```
You must create a new database and set your database user and password on your local env

**Comment by nxtreaming** [comment]
Thanks
I created the DB according to your instructions, 
Now, 'php artisan migrate' is OK

But I faced a new error (miss region)  when I run "curl http://127.0.0.1:8000"

root@ecc01:~# curl http://127.0.0.1:8000
<!DOCTYPE html>
<html lang="en" class="auto">
<!--
Spatie\LaravelIgnition\Exceptions\ViewException: Missing required client configuration options:

region: (string)

  A &quot;region&quot; configuration value is required for the &quot;s3&quot; service
  (e.g., &quot;us-west-2&quot;). A list of available public regions and endpoints can be
  found at http://docs.aws.amazon.com/general/latest/gr/rande.html. in file /usr/local/nginx/zhuanlu_html/huiwen/vendor/aws/aws-sdk-php/src/ClientResolver.php on line 454

**Comment by nxtreaming** [comment]
![image](https://user-images.githubusercontent.com/3108107/224099117-5adbd499-ef23-45cf-85a8-c60fd10a1539.png)

**Comment by nxtreaming** [comment]
I got the same error if I deployed the project to a public domain

![image](https://user-images.githubusercontent.com/3108107/224099660-5baccc83-cd40-4513-be29-dabd7168f64b.png)

**Comment by nxtreaming** [comment]
You can verify the error from:  https://zhuan.lu/huiwen/public/

**Comment by vmartuniyk** [comment]
@nxtreaming I think problem in you need feel variables on .env file
```
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=
AWS_BUCKET=
AWS_USE_PATH_STYLE_ENDPOINT=
```

**Comment by nxtreaming** [comment]
@vmartuniyk  it should be another issue , I have configured AWS in .env file
such as:

![image](https://user-images.githubusercontent.com/3108107/224201292-8560095c-4675-4f7f-8112-4d4e7dc90108.png)

**Comment by jocahdj** [comment]
He uses a 'do' disk, check config/filesystems.php. Adjust your .env variables to these and see if it works then.

![image](https://user-images.githubusercontent.com/5934805/224272848-377e9568-e07f-4bc3-a1c0-830cbeb9d256.png)

**Comment by nxtreaming** [comment]
@jocahdj  Yes, it is caused by 'do' storage, I disable it 

I finally made the project working ( https://zhuan.lu/ ). (It does not support aws S3 currently).

**Comment by crystoneme** [comment]
> @jocahdj Yes, it is caused by 'do' storage, I disable it
> 
> I finally made the project working ( https://zhuan.lu/ ). (It does not support aws S3 currently).

How to disable the "do" storage? I have the same problem

btw, I want to run locally

**Comment by nxtreaming** [comment]
@crystoneme  I just removed them, and made a small hack. (it is not recommended), I will fix this issue in a week

Please check the following diff:
https://www.dropbox.com/s/rnxndfjxwjgso6b/zhuanlu_2023-03-10.diff

**Comment by xyh2132** [comment]
> I got the same error if I deployed the project to a public domain如果我将项目部署到公共域中，我会得到相同的错误
> 
> ![image](https://user-images.githubusercontent.com/3108107/224099660-5baccc83-cd40-4513-be29-dabd7168f64b.png)

I encountered the same problem, how did you solve it?

**Comment by xyh2132** [comment]
> > @jocahdj Yes, it is caused by 'do' storage, I disable it @jocahdj 是的，这是由“do”存储引起的，我将其禁用
> > I finally made the project working ( https://zhuan.lu/ ). (It does not support aws S3 currently).我最终使项目工作了（ https://zhuan.lu/ ）。 （它目前不支持aws S3）。
> 
> How to disable the "do" storage? I have the same problem如何禁用“do”存储？我有同样的问题
> 
> btw, I want to run locally顺便说一下，我想在本地运行

Can you tell me how you solved it, please? It's the same problem.

**Comment by nxtreaming** [comment]
Just remove the codes which report errors
